### PR TITLE
virttest: add a new block.py file and write in a function get_block_virtual_size()

### PR DESCRIPTION
**Pull request by peixiu**
Create a new block.py file, related block functions or classes can write in this file.
Some scenario need to use the disk virtual size timly,
Here define a get_block_virtual_size() to get.

id: 1411207 1390485
Signed-off-by: Peixiu Hou <phou@redhat.com>

### CODE REVIEW

**Comment by peixiu** [comment]
@XueqiangWei @xiagao could you help to review this patch? thanks a lot~

File: virttest/block.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+"""
+
+from . import qemu_monitor
+
+
+def get_block_virtual_size(dev_dict):
+    """
+    Get block virtual-size of specified block device
+    from monitor's info block command.
+
```
Comment (xiagao) [inline virttest/block.py]: From the code, I believe you want to use qmp but not hmp, so pls describe it here.

File: virttest/block.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    :param dev_dict: define device dict from block info.
+    :return: Matched block virual size.
+    """
+    blocks_info = qemu_monitor.info("block")
+    for dict_block in blocks_info:
+        if dev_dict['device'] in dict_block['device']:
```
Comment (xiagao) [inline virttest/block.py]: why not use "==" ?

File: virttest/block.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+            size = dict_block['inserted']['image']['virtual-size']
+    if size:
+        return size
```
Comment (xiagao) [inline virttest/block.py]: It's better to handle size=none situation

Comment (peixiu) [reply virttest/block.py]: Hi xiagao, has updated code, could you help to review it again?

File: virttest/block.py (hunk 1 above)
Comment (xiagao) [reply virttest/block.py]: Get block virtual-size of specified block device with dev_name  ***from the qemu monitor***

Comment (peixiu) [reply virttest/block.py]: is you mean from qmp monitor?

File: virttest/block.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    :param dev_name: block device name, like "drive-hotadd" etc.
+    :return: Matched block device virual size.
+    """
+    blocks_info = qemu_monitor.info("block")
+    for dict_block in blocks_info:
+        if dev_name == dict_block['device']:
+            size = dict_block['inserted']['image']['virtual-size']
+            return int(size)
+    return None
```
Comment (XueqiangWei) [inline virttest/block.py]: it will return None by default, so I think no need to add \"return None\" in the end

Comment (peixiu) [reply virttest/block.py]: oo, thanks, will update it~

File: virttest/block.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+"""
+Utility functions to handle block related actions.
+
```
Comment (hereischen) [inline virttest/block.py]: There is a ci check error, please fix it.
if it is a one line doc string, please put it this way,
`"""Utility functions to handle block related actions."""`

File: virttest/block.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+from . import qemu_monitor
+
+
+def get_block_virtual_size(dev_name):
+    """
+    Get block virtual-size of specified block device with dev_name.
+
+    :param dev_name: block device name, like "drive-hotadd" etc.
+    :return: Matched block device virual size.
```
Comment (hereischen) [inline virttest/block.py]: Please add measurement unit of the size, eg. M, G

File: virttest/block.py (hunk 5 above)
Comment (peixiu) [reply virttest/block.py]: ok, thanks, will upodate~

File: virttest/block.py (hunk 6 above)
Comment (peixiu) [reply virttest/block.py]: ok, thanks, will update

**Comment by miyachen** [comment]
@YongxueHong Hi Yongxue, please help review this pr from power perspective, thanks.

File: virttest/block.py [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+             The default unit of the size is Byte.
+             eg: disk image is 30G, should return 32212254720.
+    """
+    blocks_info = qemu_monitor.info("block")
+    for dict_block in blocks_info:
+        if dev_name == dict_block['device']:
+            size = dict_block['inserted']['image']['virtual-size']
+            return int(size)
```
Comment (YongxueHong) [inline virttest/block.py]: Suggest this function should be move to tp-qemu/provider,  I think it is belong to case level. this function just only get the size of image, so single..

**Comment by peixiu** [comment]
Hi all, this function can be replaced by follows commands, can reduce a new function here.
    img = QemuImg(data_image_params, data_dir.get_data_dir(), data_image)
    block_virtual_size = json.loads(img.info(force_share=True,
                                    output="json"))["virtual-size"]
I added these codes to patch https://github.com/autotest/tp-qemu/pull/1302.
Closed this patch~

**Pull request closed without merging**